src/rx_signal_processing.py

```diff
@@ -115,41 +138,46 @@ def sequence_worker(options, ringbuffer):
     inproc_socket.connect("inproc://sqn_worker")
 
     while True:
-        kwargs = inproc_socket.recv_pyobj()
+        rx_params = inproc_socket.recv_pyobj()
         # Wait until kwargs received from main thread
 
-        sequence_num = kwargs["sequence_num"]
-        main_beam_angles = kwargs["main_beam_angles"]
-        intf_beam_angles = kwargs["intf_beam_angles"]
-        mixing_freqs = kwargs["mixing_freqs"]
-        slice_details = kwargs["slice_details"]
-        start_sample = kwargs["start_sample"]
-        end_sample = kwargs["end_sample"]
-        processed_data = kwargs["processed_data"]
-        intf_antennas = kwargs["intf_antennas"]
-        filter_taps = kwargs["filter_taps"]
-        downsample_rates = kwargs["downsample_rates"]
-        cfs_scan_flag = kwargs["cfs_scan_flag"]
-        samples_needed = kwargs["samples_needed"]
-        rx_rate = kwargs["rx_rate"]
-        output_sample_rate = kwargs["output_sample_rate"]
-
-        seq_begin_iden = options.dspbegin_to_brian_identity + str(sequence_num)
-        seq_end_iden = options.dspend_to_brian_identity + str(sequence_num)
-        if cfs_scan_flag:
+        # sequence_num = kwargs["sequence_num"]
+        # main_beam_angles = kwargs["main_beam_angles"]
+        # intf_beam_angles = kwargs["intf_beam_angles"]
+        # mixing_freqs = kwargs["mixing_freqs"]
+        # slice_details = kwargs["slice_details"]
+        # start_sample = kwargs["start_sample"]
+        # end_sample = kwargs["end_sample"]
+        # processed_data = kwargs["processed_data"]
+        # intf_antennas = kwargs["intf_antennas"]
+        # filter_taps = kwargs["filter_taps"]
+        # downsample_rates = kwargs["downsample_rates"]
+        # cfs_scan_flag = kwargs["cfs_scan_flag"]
+        # samples_needed = kwargs["samples_needed"]
+        # rx_rate = kwargs["rx_rate"]
+        # output_sample_rate = kwargs["output_sample_rate"]
+
+        seq_begin_iden = options.dspbegin_to_brian_identity + str(
+            rx_params.sequence_num
+        )
+        seq_end_iden = options.dspend_to_brian_identity + str(rx_params.sequence_num)
+        if rx_params.cfs_scan_flag:
             sender_iden = options.dsp_cfs_identity
             recipient_iden = options.radctrl_cfs_identity
         else:
-            sender_iden = options.dsp_to_dw_identity + str(sequence_num)
+            sender_iden = options.dsp_to_dw_identity + str(rx_params.sequence_num)
             recipient_iden = options.dw_to_dsp_identity
         log.info(
             "socket identities:",
             sender=sender_iden,
             recip=recipient_iden,
-            cfs_flag=cfs_scan_flag,
+            cfs_flag=rx_params.cfs_scan_flag,
         )
         sequence_worker_sockets = so.create_sockets(
```

Comment:
Here you could capture the return as `dspbegin_to_brian, dspend_to_brian, processed_socket = so.create_sockets(...`. This was my original idea for changing the signature. Very bike-shedy though, no need to actually change it

Reply:
Not entirely following here so I'll leave it for now